app/common/data/models_user.py

```diff
@@ -30,6 +30,57 @@ class User(BaseModel):
     roles: Mapped[list["UserRole"]] = relationship("UserRole", back_populates="user", cascade="all, delete-orphan")
     submissions: Mapped[list["Submission"]] = relationship("Submission", back_populates="created_by")
 
+    # These relationships need Organisation table as part of the join for filtering
+    # Using string expressions to allow forward references to work properly
+    # NOTE: This does not account for platform admin privileges; this is specifically for normal users
+    deliver_grants: Mapped[list["Grant"]] = relationship(
+        "Grant",
+        secondary="join(UserRole, Organisation, UserRole.organisation_id == Organisation.id)",
+        primaryjoin="User.id == UserRole.user_id",
+        secondaryjoin="""and_(
+            or_(
+                Grant.id == UserRole.grant_id,
+                and_(
+                    Grant.organisation_id == UserRole.organisation_id,
+                    UserRole.grant_id.is_(None)
+                )
+            ),
+            Organisation.can_manage_grants == True
+        )""",
+        viewonly=True,
+        order_by="Grant.name",
+    )
+
+    # For access_grants, we need to join through GrantRecipient for organization-level access
+    # The logic is:
+    # - Direct: Grant.id == UserRole.grant_id, OR
+    # - Org-level: exists(GrantRecipient where GrantRecipient.grant_id == Grant.id
+    #              AND GrantRecipient.organisation_id == UserRole.organisation_id)
+    # We also filter by Organisation.can_manage_grants == False
+    # NOTE: This does not account for platform admin privileges; this is specifically for normal users
+    access_grants: Mapped[list["Grant"]] = relationship(
```

Comment:
I wonder if we'll have to do some thinking about a consistent way of keeping properties like this useful in unit test factories (in general not this change because I think on the ORM model like this is a way of keeping database business logic neatly in one place) - there's one place on components we had to encapsulate what the join _should_ have been doing with an underlying database. 

I think in most cases this won't matter as the unit test itself can take care of it if it actually matters to that test but some of these concepts might find themselves solving the same issue in a few places (what we found on components)